File: Android-master/Android-master/HW 1-3/contacts/src/main/java/com/example/contacts/AddContactActivity.java
package com.example.contacts;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.text.InputType;
import android.view.MenuItem;
import android.view.View;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.RadioButton;

public class AddContactActivity extends AppCompatActivity {

    private RadioButton radioPhone;
    private RadioButton radioEmail;
    private EditText nameField;
    private EditText contactField;
    private ImageView save;
    private Toolbar toolbar;
    private ActionBar actionBar;
    private Contact contact;
    public static final String KEY_NEW_CONTACT = "KEY_NEW_CONTACT";
    public static final int REQUEST_CODE_ADD_CONTACT = 111;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.add_contact);

        radioPhone = findViewById(R.id.radio_phoneNumber);
        radioEmail = findViewById(R.id.radio_email);
        nameField = findViewById(R.id.personName_field);
        contactField = findViewById(R.id.personContact_field);
        save = findViewById(R.id.saveButton);
        toolbar = findViewById(R.id.toolbarAddActivity);
        setSupportActionBar(toolbar);

        radioPhone.setOnClickListener(radioButtonListener);
        radioEmail.setOnClickListener(radioButtonListener);
        save.setOnClickListener(saveButtonListener);

        actionBar = getSupportActionBar();
        if(actionBar != null){
            actionBar.setDisplayHomeAsUpEnabled(true);
        }
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            finish();
        }
        return super.onOptionsItemSelected(item);
    }

    public static Intent newIntent(Context context){
        Intent intent = new Intent(context, AddContactActivity.class);
        return intent;
    }

    View.OnClickListener radioButtonListener = new View.OnClickListener() {
        @Override
        public void onClick(View view) {
            switch (view.getId()){
                case R.id.radio_phoneNumber:
                    contactField.setHint("Phone number");
                    contactField.setInputType(InputType.TYPE_CLASS_PHONE);
                    break;
                case R.id.radio_email:
                    contactField.setHint("Email");
                    contactField.setInputType(InputType.TYPE_TEXT_VARIATION_EMAIL_ADDRESS);
                    break;
                default:
                    break;
            }
        }
    };

    View.OnClickListener saveButtonListener = new View.OnClickListener() {
        @Override
        public void onClick(View view) {
            contact = createContact(nameField.getText().toString(), contactField.getText().toString());
            Intent resultIntent = new Intent();
            resultIntent.putExtra(KEY_NEW_CONTACT, contact);
            setResult(Activity.RESULT_OK, resultIntent);
            finish();
        }
    };

    private Contact createContact(String nameField, String contactField){
        if(radioPhone.isChecked()){
            contact = new Contact.Builder()
                    .setPersonName(nameField)
                    .setPhone(contactField)
                    .build();
        } else if(radioEmail.isChecked()){
            contact = new Contact.Builder()
                    .setPersonName(nameField)
                    .setEmail(contactField)
                    .build();
        }return contact;
    }
}

File: HW 5/kotlin_sql/src/main/java/com/example/kotlin_sql/AddContactActivity.kt
package com.example.kotlin_sql

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.text.InputType
import android.view.MenuItem
import android.view.View
import androidx.appcompat.app.ActionBar
import androidx.appcompat.app.AppCompatActivity
import com.example.kotlin_sql.CONTACT_TYPE.EMAIL
import com.example.kotlin_sql.CONTACT_TYPE.PHONE
import kotlinx.android.synthetic.main.add_contact.*

const val KEY_NEW_CONTACT = "KEY_NEW_CONTACT"

class AddContactActivity : AppCompatActivity() {

    private var actionBar: ActionBar? = null
    private var contact: Contact? = null

    fun newIntent(context: Context?): Intent? {
        return Intent(context, AddContactActivity::class.java)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.add_contact)

        setSupportActionBar(toolbarAddActivity)

        radioPhone?.setOnClickListener(radioButtonListener)
        radioEmail?.setOnClickListener(radioButtonListener)
        saveButton?.setOnClickListener(saveButtonListener)

        actionBar = supportActionBar
        actionBar?.setDisplayHomeAsUpEnabled(true)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
        }
        return super.onOptionsItemSelected(item)
    }

    private var radioButtonListener = View.OnClickListener { view ->
        when (view.id) {
            R.id.radioPhone -> adjustPhoneField()
            R.id.radioEmail -> adjustEmailField()
        }
    }

    private fun adjustPhoneField() {
        personContactField?.hint = "Phone number"
        personContactField?.inputType = InputType.TYPE_CLASS_PHONE
    }

    private fun adjustEmailField() {
        personContactField?.hint = "Email"
        personContactField?.inputType = InputType.TYPE_TEXT_VARIATION_EMAIL_ADDRESS
    }

    private val saveButtonListener = View.OnClickListener {
        contact = createContact(personNameField?.text.toString(), personContactField?.text.toString())
        val resultIntent = Intent()
        resultIntent.putExtra(KEY_NEW_CONTACT, contact)
        setResult(RESULT_OK, resultIntent)
        finish()
    }

    private fun createContact(nameField: String, contactField: String): Contact{
        if (radioPhone.isChecked) {
            contact = Contact(nameField, PHONE, contactField)
        } else if (radioEmail.isChecked) {
            contact = Contact(nameField, EMAIL, contactField)
        }
        return contact!!
    }
}